datafusion/expr/src/type_coercion.rs

```diff
@@ -15,247 +15,49 @@
 // specific language governing permissions and limitations
 // under the License.
 
-//! Type coercion rules for functions with multiple valid signatures
+//! Type coercion rules for DataFusion
 //!
 //! Coercion is performed automatically by DataFusion when the types
-//! of arguments passed to a function do not exacty match the types
-//! required by that function. In this case, DataFusion will attempt to
-//! *coerce* the arguments to types accepted by the function by
-//! inserting CAST operations.
+//! of arguments passed to a function or needed by operators do not
+//! exacty match the types required by that function / operator. In
+//! this case, DataFusion will attempt to *coerce* the arguments to
+//! types accepted by the function by inserting CAST operations.
 //!
 //! CAST operations added by coercion are lossless and never discard
-//! information. For example coercion from i32 -> i64 might be
+//! information.
+//!
+//! For example coercion from i32 -> i64 might be
 //! performed because all valid i32 values can be represented using an
 //! i64. However, i64 -> i32 is never performed as there are i64
 //! values which can not be represented by i32 values.
-//!
-
-use crate::{Signature, TypeSignature};
```

Comment:
Moved this code into `Moved to `type_coercion/function.rs` as it is related to function coercion, despite the `data_types` generic name